src/Resolution.ts

```diff
@@ -49,160 +43,116 @@ import DnsUtils from './DnsUtils';
  * ```
  */
 export default class Resolution {
-  /** @internal */
-  readonly blockchain: boolean;
-  /** @internal */
-  readonly ens?: Ens;
-  /** @internal */
-  readonly zns?: Zns;
-  /** @internal */
-  readonly cns?: Cns;
-  /** @internal */
-  readonly api?: UdApi;
-
+  readonly serviceMap: Record<NamingServiceName, NamingService>;
+  
   constructor({
-    blockchain = true,
-    api = DefaultAPI,
-  }: { blockchain?: Blockchain | boolean; api?: API } = {}) {
-    this.blockchain = !!blockchain;
-    if (blockchain) {
-      if (blockchain === true) {
-        blockchain = {};
-      }
-
-      const web3provider = blockchain.web3Provider;
-      if (web3provider) {
-        console.warn(
-          'Usage of `web3Provider` option is deprecated. Use `provider` option instead for each individual blockchain',
-        );
-      }
-
-      const ens = this.normalizeSource(blockchain.ens, web3provider);
-      const zns = this.normalizeSource(blockchain.zns);
-      const cns = this.normalizeSource(blockchain.cns, web3provider);
-
-      if (ens) {
-        this.ens = new Ens(ens);
-      }
-
-      if (zns) {
-        this.zns = new Zns(zns);
-      }
-
-      if (cns) {
-        this.cns = new Cns(cns);
-      }
+    sourceConfig = undefined,
+  }: { sourceConfig?: SourceConfig } = {}) {
 
-    } else {
-      this.api = new UdApi(api);
-    }
+    const cns = (isApi(sourceConfig?.cns) ? new UdApi() : new Cns(sourceConfig?.cns));
+    const ens = (isApi(sourceConfig?.ens) ? new UdApi() : new Ens(sourceConfig?.ens));
+    const zns = (isApi(sourceConfig?.zns) ? new UdApi() : new Zns(sourceConfig?.zns));
 
+    this.serviceMap = {
+      [NamingServiceName.CNS]: cns,
+      [NamingServiceName.ENS]: ens,
+      [NamingServiceName.ZNS]: zns,
+    };
   }
 
   /**
    * Creates a resolution with configured infura id for ens and cns
-   * @param infura infura project id
-   * @param network ethereum network name
+   * @param infura - infura project id
+   * @param networks - an optional object that describes what network to use when connecting ENS or CNS default is mainnet
```

Comment:
> an optional object that describes what network to use when connecting ENS or CNS default is mainnet

I'm not sure I understand what this comment is saying

Reply:
I've unresolved this because I still believe there's some problem with the grammar or wording. I believe it'll start making sense if we put a comma before the word "default".